rclcpp/src/rclcpp/publisher.cpp

```diff
@@ -231,6 +230,19 @@ PublisherBase::setup_intra_process(
   IntraProcessManagerSharedPtr ipm,
   const rcl_publisher_options_t & intra_process_options)
 {
+  // Skip intraprocess configuration if the "durability" qos policy is not "volatile".
+  // TODO(ivanpauno):
+  //    RMW_QOS_POLICY_DURABILITY_SYSTEM_DEFAULT is for FastRTPS and Connext also
+  //    equal to "volatile". It would be good to add a
+  //    is_durability_policy_volatile(rcl_publisher_options_t options)
+  //    method to rcl, and corresponging methods in rmw and the rmw vendor
+  //    implementations, which will also checks if "system_default" is "volatile" or not.
```

Comment:
But isn't this exactly what your other pr's are aiming to do?

Reply:
I will update when I merge them

Reply:
Sorry that I wasn't clear. But I'm not thinking to merge this before the other PRs. I'm now going to update this.